addons/arsenal/functions/fnc_scanConfig.sqf

```diff
@@ -12,178 +13,223 @@
  * Public: No
 */
 
-private _cargo = [
-    [[], [], []], // Weapons 0, primary, secondary, handgun
-    [[], [], [], []], // WeaponAccessories 1, optic,side,muzzle,bipod
-    [ ], // Magazines 2
-    [ ], // Headgear 3
-    [ ], // Uniform 4
-    [ ], // Vest 5
-    [ ], // Backpacks 6
-    [ ], // Goggles 7
-    [ ], // NVGs 8
-    [ ], // Binoculars 9
-    [ ], // Map 10
-    [ ], // Compass 11
-    [ ], // Radio slot 12
-    [ ], // Watch slot  13
-    [ ], // Comms slot 14
-    [ ], // WeaponThrow 15
-    [ ], // WeaponPut 16
-    [ ] // InventoryItems 17
-];
-
-private _configCfgWeapons = configFile >> "CfgWeapons"; //Save this lookup in variable for perf improvement
+private _configItems = EMPTY_VIRTUAL_ARSENAL;
 
+private _filterFunction = [{
+    isClass _x && {(if (isNumber (_x >> 'scopeArsenal')) then {getNumber (_x >> 'scopeArsenal')} else {getNumber (_x >> 'scope')}) == 2} && {getNumber (_x >> 'ace_arsenal_hide') != 1}
```

Comment:
Should items with `scope` < 2 and `scopeArsenal` = 2 be allowed?

Reply:
Addressed in [4c9eb8b](https://github.com/acemod/ACE3/pull/9040/commits/4c9eb8b480d2dc32c95f150a8fd040f6cc527af6).